src/txn/txn_ckpt.c

```diff
@@ -2021,8 +2023,8 @@ __checkpoint_lock_dirty_tree(
                 skip_ckpt = false;
         }
 
-        /* Skip the clean btree until the btree has obsolete pages. */
-        if (skip_ckpt && !F_ISSET(btree, WT_BTREE_OBSOLETE_PAGES)) {
+        if (skip_ckpt &&
+          !F_ISSET(btree, WT_BTREE_OBSOLETE_PAGES && !bm->can_truncate(btree->bm, session))) {
```

Comment:
I think the `()` are wrong:
```suggestion
        if (skip_ckpt &&
          !F_ISSET(btree, WT_BTREE_OBSOLETE_PAGES) && !bm->can_truncate(btree->bm, session)) {
```

Reply:
Good point! Thank you